cvat-sdk/cvat_sdk/core/uploading.py

```diff
@@ -341,43 +345,99 @@ def upload_files(
+                        url,
+                        post_params=dict(**kwargs, **files),
+                        headers={
+                            "Content-Type": "multipart/form-data",
+                            "Upload-Multiple": "",
+                            **self._client.api_client.get_common_headers(),
+                        },
+                    )
+                expect_status(200, response)
 
-        self._tus_start_upload(url)
+                if pbar is not None:
+                    pbar.advance(group_size)
 
-        for group, group_size in bulk_file_groups:
-            with ExitStack() as es:
-                files = {}
-                for i, filename in enumerate(group):
-                    files[f"client_files[{i}]"] = (
-                        filename,
-                        es.enter_context(closing(open(filename, "rb"))).read(),
-                    )
-                response = self._client.api_client.rest_client.POST(
+            for filename in separate_files:
+                # TODO: check if basename produces invalid paths here, can lead to overwriting
+                self._upload_file_data_with_tus(
                     url,
-                    post_params=dict(**kwargs, **files),
-                    headers={
-                        "Content-Type": "multipart/form-data",
-                        "Upload-Multiple": "",
-                        **self._client.api_client.get_common_headers(),
-                    },
+                    filename,
+                    meta={"filename": osp.basename(filename)},
+                    pbar=pbar,
+                    logger=self._client.logger.debug,
                 )
-            expect_status(200, response)
 
-            if pbar is not None:
-                pbar.advance(group_size)
-
-        for filename in separate_files:
-            # TODO: check if basename produces invalid paths here, can lead to overwriting
-            self._upload_file_data_with_tus(
-                url,
-                filename,
-                meta={"filename": osp.basename(filename)},
-                pbar=pbar,
-                logger=self._client.logger.debug,
-            )
+            self._tus_finish_upload(url, fields=kwargs)
 
-        self._tus_finish_upload(url, fields=kwargs)
+    def _has_upload_metainfo_file(self, resources: Sequence[str]) -> bool:
+        file = self._find_upload_metainfo_file(resources)
+        return file and osp.isfile(file)
+
+    def _find_upload_metainfo_file(self, resources: Sequence[str]) -> Optional[str]:
+        candidates = [f for f in resources if f.endswith(".jsonl")]
+        if not candidates:
+            return None
+        elif len(candidates) != 1:
+            raise TooManyManifests(candidates=candidates)
+        else:
+            return candidates[0]
+
+    @staticmethod
+    def _default_upload_metainfo_filename() -> str:
+        return "manifest.jsonl"
```

Comment:
This seems like it should be a constant.

Reply:
Not actual anymore